os/fs/smartfs/smartfs_utils.c

```diff
@@ -849,7 +858,11 @@ int smartfs_finddirentry(struct smartfs_mountpt_s *fs, struct smartfs_entry_s *d
 			seglen++;
 			ptr++;
 		}
-
+		/* Perform a check to avoid bufer overflow */
+		if (seglen >= 256) {
```

Comment:
actually workerbuffer is allocated at mount, it also uses constant value. i think we can make definition in smartfs.h
#define SMARTFS_WORKERBUFFER_LEN or something.

Reply:
Yes, always better to use macro.
Will update when we implement solution to stat fail issue.